src/ibidem.rs

```diff
@@ -71,15 +73,30 @@ impl<P: syn::parse::Parse> std::ops::Deref for ParseWrapper<P> {
 
 /// While it is convenient to be able to "pass through" TokenStreams
 /// unperturbed, or to interpret TokenStreams via syn::parse::Parse.
+///
 /// While serde--wisely--does not permit this kind of unholy communion between
 /// Deserialize and Deserializer, we can skirt around this with the
 /// otherwise-unused deserialize_bytes/visit_bytes interfaces. Since there is
 /// no TokenStream that could reasonably be interpreted as bytes, we use this
-/// interface as a conduit to pass through a serialized form of a portion of
-/// the TokenStream from our Deserializer::deserialize_bytes() and deserialize
-/// it in WrapperVisitor::visit_bytes(). Yes, there is a serializer/deserializer
-/// pair buried *within* the broader deserialization. Gotta spend money to make
-/// money.
+/// interface to signal to the `serde_tokenstream` deserializer that we should
+/// be interpreting the TokenStream directly.
+///
+/// The signal works via a thread-local storage (TLS) side channel. When we
+/// want to interpret a TokenStream directly:
+///
+/// 1. First, `TokenStreamWrapper` or `ParseWrapper` calls
+///    `deserializer.deserialize_bytes(WrapperVisitor)`, assuming that
+///    `deserializer` is always the `serde_tokenstream` deserializer.
+/// 2. The `serde_tokenstream` deserializer calls `set_wrapper_tokens` with the
+///    `TokenStream`.
+/// 3. The `serde_tokenstream` deserializer calls `visitor.visit_bytes`, which
+///    is always the `WrapperVisitor`.
+/// 4. The `WrapperVisitor` deserializer immediately calls
+///    `take_wrapper_tokens` to retrieve the `TokenStream`.
+///
+/// Previously, we serialized to bytes and later deserialized them, but that
+/// lost span information. Storing the actual tokens directly preserves it, and
+/// this benefit probably outweighs the ick of using TLS/dynamic scoping.
```

Comment:
```suggestion
/// So, yes: this is ick. However, it allows us to retain Span information which in turn allows
/// us to craft very good, targeted errors to guide users in the case of bad input.
```

I generally prefer to let the git history be the record of past implementations, so what you think of just explaining the benefit of this approach?

Reply:
Sure, that's fine. I guess for me the "allows us to retain Span information" doesn't entirely logically connect on its own, so I'll add some other text like "unlike alternatives like serializing the `TokenStream` as bytes".